src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/FileDialog.cs

```diff
@@ -1932,12 +1978,16 @@ private static IShellItem ResolveCustomPlace(FileDialogCustomPlace customPlace)
         //---------------------------------------------------
         #region Private Fields
 
-        private const FOS c_VistaFileDialogMask = FOS.OVERWRITEPROMPT | FOS.NOCHANGEDIR | FOS.NOVALIDATE | FOS.ALLOWMULTISELECT | FOS.PATHMUSTEXIST | FOS.FILEMUSTEXIST | FOS.CREATEPROMPT | FOS.NODEREFERENCELINKS;
+        private const FOS c_VistaFileDialogMask = FOS.OVERWRITEPROMPT | FOS.NOCHANGEDIR | FOS.NOVALIDATE | FOS.ALLOWMULTISELECT | FOS.PATHMUSTEXIST | FOS.FILEMUSTEXIST | FOS.CREATEPROMPT | FOS.NODEREFERENCELINKS | FOS.PICKFOLDERS;
```

Comment:
There is a comment on `Options`:

```
//   We bitwise AND _dialogOptions with all of the options we consider valid
//   before returning the resulting bitmask to avoid accidentally setting a
//   flag we don't intend to. [...]
```

I was trying to preserve this concern and not sneak in any unrelated behavior changes. However, it is this design that prevented even using reflection to set the PICKFOLDERS flag, and I did not consider the possibility that it was due to the shared flags storage. Since the `SetFlags` is internal, I agree we could lift the masking and will update accordingly.